sharding/indexHashedNodesCoordinator_test.go

```diff
@@ -532,6 +653,208 @@ func BenchmarkIndexHashedGroupSelector_ComputeValidatorsGroup21of400(b *testing.
 	}
 }
 
+func BenchmarkIndexHashedGroupSelector_ComputeValidatorsGroup63of400RecomputeEveryGroup(b *testing.B) {
+	consensusGroupSize := 63
+	list := make([]sharding.Validator, 0)
+
+	//generate 400 validators
+	for i := 0; i < 400; i++ {
+		list = append(list, mock.NewValidatorMock(big.NewInt(0), 0, []byte("pk"+strconv.Itoa(i)), []byte("addr"+strconv.Itoa(i))))
+	}
+
+	nodesMap := make(map[uint32][]sharding.Validator)
+	nodesMap[0] = list
+
+	consensusGroupCache, _ := lrucache.NewCache(10000)
+
+	arguments := sharding.ArgNodesCoordinator{
+		ShardConsensusGroupSize: consensusGroupSize,
+		MetaConsensusGroupSize:  1,
+		Hasher:                  &mock.HasherMock{},
+		NbShards:                1,
+		Nodes:                   nodesMap,
+		SelfPublicKey:           []byte("key"),
+		ConsensusGroupCache:     consensusGroupCache,
+	}
+	ihgs, _ := sharding.NewIndexHashedNodesCoordinator(arguments)
+
+	m := runtime.MemStats{}
+	runtime.ReadMemStats(&m)
+
+	fmt.Println(m.HeapAlloc)
+
+	//missedBlocks := 1000
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		randomness := strconv.Itoa(i)
+		list2, _ := ihgs.ComputeValidatorsGroup([]byte(randomness), uint64(i), 0)
+		assert.Equal(b, consensusGroupSize, len(list2))
+	}
+
+	m2 := runtime.MemStats{}
+
+	runtime.ReadMemStats(&m2)
```

Comment:
This doesn't work as expected because the benchmarking process does one iteration to count the number of ns required and then approximates how many runs are needed in one second to compute b.N and after that calls the Benchmarking function again.

Reply:
I did change it. Hopefully now it should work now.